sdk/src/main/java/io/customer/sdk/repository/AttributesRepository.kt

```diff
@@ -0,0 +1,34 @@
+package io.customer.sdk.repository
+
+import com.squareup.moshi.JsonAdapter
+import java.util.*
+import kotlin.collections.HashMap
+
+internal interface AttributesRepository {
+    fun mapToJson(map: Map<String, Any>): Map<String, Any>
+}
+
+internal class MoshiAttributesRepositoryImp(
+    private val jsonAdapter: JsonAdapter<Map<String, Any>>,
+) : AttributesRepository {
+
+    override fun mapToJson(map: Map<String, Any>): Map<String, Any> {
+        return jsonAdapter.fromJsonValue(verifyMap(map)) ?: emptyMap()
+    }
+
+    private fun verifyMap(map: Map<String, Any>): Map<String, Any> {
+        val validMap = HashMap<String, Any>()
+        map.entries.forEach {
+            validMap[it.key] = getValidValue(it.value)
+        }
+        return validMap.toMap()
+    }
+
+    // Unix timestamp date are only acceptable
+    private fun getValidValue(any: Any): Any {
+        return when (any) {
+            is Date -> any.time
```

Comment:
I just ran into this issue today on iOS: make sure that `any.time` does not have decimal values. The API does not accept dates with a decimal. Decimal values may exist if the Date object has milliseconds.

Reply:
Ack

Reply:
Just confirmed, `any.time` covers into `Long` and doesn't have any decimal values.